test/Garnet.test/Extensions/TxnCustomCmd.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Garnet.common;
+using Garnet.server;
+using Tsavorite.core;
+
+namespace Garnet
+{
+    class TxnCustomCmd : CustomTransactionProcedure
+    {
+        public override bool Prepare<TGarnetReadApi>(TGarnetReadApi api, ref CustomProcedureInput procInput)
+        {
+            var offset = 0;
+
+            var mainStoreKey = GetNextArg(ref procInput, ref offset);
+            _ = GetNextArg(ref procInput, ref offset); // mainStoreValue
+
+            AddKey(mainStoreKey, LockType.Exclusive, false);
+
+            var myDictKey = GetNextArg(ref procInput, ref offset);
+            AddKey(myDictKey, LockType.Exclusive, true);
+
+            return true;
+        }
+
+        public override void Main<TGarnetApi>(TGarnetApi api, ref CustomProcedureInput procInput, ref MemoryResult<byte> output)
+        {
+            var offset = 0;
+
+            var mainStoreKey = GetNextArg(ref procInput, ref offset);
+            var mainStoreValue = GetNextArg(ref procInput, ref offset);
+
+            api.SET(mainStoreKey, mainStoreValue);
+
+            var myDictKey = GetNextArg(ref procInput, ref offset);
+            var myDictField = GetNextArg(ref procInput, ref offset);
+            var myDictValue = GetNextArg(ref procInput, ref offset);
+
+            var args = new ArgSlice[2];
+            args[0] = myDictField;
+            args[1] = myDictValue;
+
+            ExecuteCustomCommand(api, "MYDICTSET", myDictKey, args, out var _output);
```

Comment:
Looks like this will have to match and retrieve the command every single time main is called, which is on the critical path. Users should be able to, in Prepare, get the actual ID so that the critical path call is very light.